run.py

```diff
@@ -107,6 +111,44 @@
 NEXT_FILE_TOKEN = '<nextfile>'
 
 
+class SoftLabelCrossEntropyLoss(nn.Module):
```

Comment:
What's the difference between `SoftLabelCrossEntropyLoss` and `torch.nn.CrossEntropyLoss`?

Reply:
`nn.CrossEntropyLoss` is optimized for hard labels, i.e., if the target distribution is 1.0 for one class and zero for all others. I guess `nn.BCELoss` would also work; mathematically slightly different, not sure if it matters...

Reply:
Alright, I see, `nn.BCELoss` uses the sigmoid function, which is better for multilabel classification? 

This is what I had before.

Reply:
BCELoss treats each label as a separate binary variable and averages over them. So, not really a soft label by the books, but maybe close enough...would be interesting to see if it makes a difference...